backend/mcp_backend/routers/crew_mcp.py

```diff
@@ -27,45 +29,242 @@ def extract_user_token(
-        return token
+    logging.info("=== OAuth Token Extraction Debug ===")
+    logging.info(f"Available headers: {list(request.headers.keys())}")
+    logging.info(f"Available cookies: {list(request.cookies.keys())}")
+
+    # Method 1: OAuth2 proxy forwarded headers
+    # Check common OAuth2 proxy header patterns
+    oauth_headers_to_check = [
+        "X-Forwarded-Access-Token",
+        "X-Auth-Request-Access-Token",
+        "X-Oauth-Token",
+        "X-Access-Token",
+        "Authorization",
+    ]
+
+    for header_name in oauth_headers_to_check:
+        token = request.headers.get(header_name, "")
+        if token:
+            # Handle Authorization header specifically
+            if header_name == "Authorization" and token.startswith("Bearer "):
+                token = token.split("Bearer ", 1)[1]
+
+            if token and token != "user_token_placeholder":
+                logging.info(
+                    f"Found OAuth token in header {header_name} (length: {len(token)})"
+                )
+                return token
+
+    # Method 2: OAuth2 proxy cookies (standard oauth2-proxy cookie names)
+    oauth_cookies_to_check = [
+        "oauth_access_token",
+        "oauth_id_token",
+        "oauth_token",
+        "_oauth2_proxy",  # Default oauth2-proxy session cookie
+        "CANChat",  # Your configured cookie name
+    ]
+
+    for cookie_name in oauth_cookies_to_check:
+        token = request.cookies.get(cookie_name, "")
+        if token and token != "user_token_placeholder":
+            logging.info(
+                f"Found OAuth token in cookie {cookie_name} (length: {len(token)})"
+            )
 
-    # Second priority: OAuth ID token (may work for some scenarios)
-    elif "oauth_id_token" in request.cookies:
-        token = request.cookies.get("oauth_id_token")
-        logging.info("Using OAuth ID token for SharePoint OBO flow")
-        return token
+            # If it's the oauth2-proxy session cookie, try to extract the access token
+            if cookie_name in ["_oauth2_proxy", "CANChat"]:
+                try:
+                    # OAuth2 proxy session cookies are often base64 encoded JSON
+                    # Try to decode and extract access token
+                    decoded_data = base64.b64decode(token + "==").decode("utf-8")
+                    session_data = json.loads(decoded_data)
+                    if "access_token" in session_data:
+                        access_token = session_data["access_token"]
+                        logging.info(
+                            f"Extracted access token from session cookie {cookie_name}"
+                        )
+                        return access_token
+                except (ValueError, json.JSONDecodeError, UnicodeDecodeError) as e:
+                    logging.debug(f"Could not decode session cookie {cookie_name}: {e}")
+                    # Fall back to using the cookie value as-is
+                    pass
```

Comment:
Truthfully, I wouldn't use the cookie here. It's not meant to be a standard way of interfacing with oauth2-proxy and means that it's not a defined contract and may change at anytime. The cookie that is created is only there for session storage and that's only if using the cookie-based storage. This will likely not be the case down the line since we'll probably move to redis-based session storage.